client/elements/menus/sc-action-items.js

```diff
@@ -304,6 +304,13 @@ export class SCActionItems extends LitLocalized(LitElement) {
     this.#displaySearchOptionsButtonStateChange();
     this.scSiteLayout = document.querySelector('sc-site-layout');
     this.#setToolBarLanguage();
+    document.addEventListener('keydown', this._handleKeydown.bind(this));
+  }
+
+  _handleKeydown(e) {
+    if (e.key === 'Escape') {
+      this.#closeAllTopSheetsOfTextPage();
```

Comment:
While I'm here, I also added a keyboard shortcut to close all the top sheets: `esc`